<launch>
  <node name="robot_distance_monitor" pkg="robot_distance_monitor" type="robot_distance_monitor_node">
    <param name="~odom_topic" value="/ropod/odom"/>
  </node>

  <node name="robot_heading_monitor" pkg="robot_heading_monitor" type="robot_heading_monitor_node">
    <param name="~odom_topic" value="/ropod/odom"/>
  </node>

  <node name="line_extractor" pkg="laser_line_extraction" type="line_extraction_node">
    <param name="~frequency" value="30.0" />
    <param name="~frame_id" value="hokuyo_link" />
    <param name="~scan_topic" value="/ropod/laser/scan" />
    <param name="~publish_markers" value="true" />
    <param name="~bearing_std_dev" value="1e-5" />
    <param name="~range_std_dev" value="0.012" />
    <param name="~least_sq_angle_thresh" value="0.0001" />
    <param name="~least_sq_radius_thresh" value="0.0001" />
    <param name="~max_line_gap" value="0.5" />
    <param name="~min_line_length" value="0.3" />
    <param name="~min_range" value="0.5" />
    <param name="~min_split_dist" value="0.04" />
    <param name="~outlier_dist" value="0.06" />
    <param name="~min_line_points" value="10" />
  </node>

  <node name="semantic_feature_detection_node" pkg="semantic_feature_detection" type="semantic_feature_detection_node">
    <param name="~frequency" value="10.0" />
    <param name="~frame_id" value="hokuyo_link" />
    <param name="~laser_line_extraction_topic" value="/line_segments" />
  </node>

  <node name="gateway_detection_node" pkg="gateway_detection" type="gateway_detection_node">
    <param name="~frequency" value="10.0" />
    <param name="~frame_id" value="hokuyo_link" />
    <param name="~semantic_feature_detection_topic" value="/semantic_features_detected" />
  </node>

  <node name="Operator" pkg="nav2d_operator" type="operator" >
    <remap from="scan" to="/ropod/laser/scan"/>
    <rosparam file="$(find nav2d_operator)/config/operator.yaml"/>
    <rosparam file="$(find nav2d_operator)/config/costmap.yaml" ns="local_map" />
  </node>

  <node name="heading_control" pkg="heading_control" type="heading_control_node">
    <param name="~desired_heading_topic" value="/desired_heading"/>
    <param name="~monitored_heading_topic" value="/monitored_heading"/>
    <param name="~heading_control_velocity" value="0.1"/>
  </node>
</launch>